lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -547,11 +556,24 @@ void Inliner::rename(StringRef prefix, Operation *op,
         // This matters when the same module is inlined twice and the NLA only
         // applies to one of them.
         auto &mnla = nlaMap[sym.getAttr()];
-        if (!anno.isClass("circt.nonlocal") &&
-            !doesNLAMatchCurrentPath(mnla.getNLA()))
+        if (!doesNLAMatchCurrentPath(mnla.getNLA()))
           continue;
         mnla.setInnerSym(moduleNamespace.module.moduleNameAttr(), newSymAttr);
       }
+      if (instanceParent) {
+        // The InstanceOp is renamed, so move the HierPathOps to the new
+        // InnerRefAttr.
+        auto newInnerRef = InnerRefAttr::get(instanceParent, newSymAttr);
+        auto oldInnerRef = InnerRefAttr::get(instanceParent, oldInstSym);
+        instOpHierPaths[newInnerRef] = std::move(instOpHierPaths[oldInnerRef]);
```

Comment:
Consider dropping the oldInnerRef from the map after `std::move`'ing out from its storage, so the now-invalid (at best, empty) value is no longer sticking around.  Shouldn't be a problem as-is AFAIK but also helps keep map size down :wink: and invariants like "all keys have non-empty values".